lib/libmpa/mpa_random.c

```diff
@@ -35,12 +35,12 @@
 #include "mpa_assert.h"
 #include <tee_api_types.h>
 
-/* This function ends to the real RNG of the platform
- * Depending where the libmpa is used:
- * - kernel side: the implementation is found in the TEECore
- * - user side: the implementation is found in libutee
- */
-TEE_Result get_rng_array(void *buf, size_t blen);
+static random_generator_cb get_rng_array;
+
+void mpa_set_random_generator(random_generator_cb callback)
+{
+	get_rng_array = callback;
+}
```

Comment:
I think you dismissed the comment about kernel and user side a bit too quickly ;)
When libmpa is built in user mode, `get_rng_array` will never get initialized (because `mpa_set_random_generator()` is called only by the TEE core, in core/lib/libtomcrypt/src/tee_ltc_provider.c).
So I suspect `TEE_BigIntIsProbablePrime()` can assert: `TEE_BigIntIsProbablePrime()` ->  `mpa_is_prob_prime()` -> `primality_test_miller_rabin()` -> `mpa_get_random()` -> `ASSERT()`.